nexus/db-queries/src/db/datastore/vpc.rs

```diff
@@ -1603,6 +1604,12 @@ mod tests {
         }
     }
 
+    fn sled_states_active(
```

Comment:
The repetition of this reinforces the idea that we could use a nice low-level builder interface that could take care of this.

Reply:
I thought this would require some serious rework of `BlueprintBuilder` and the planner to make the builder less plan-y, but I was wrong (although I think there is still a lot of room for improvement there). I reworked this test to use BlueprintBuilder in [4949eac](https://github.com/oxidecomputer/omicron/pull/5663/commits/4949eac70c7ccfda81d5fd8dea8704f19e4a0f65) and it went pretty well. There were two spots (creating `bp2` and `bp4`) where it was easier to clone the parent blueprint and manually poke it than to go through the builder, but it still seems like a pretty significant cleanup overall.